lib/connection.js

```diff
@@ -377,6 +397,8 @@ class Connection extends EventEmitter {
       this, application, authStrategy, credentials,
       this._onSessionCreated.bind(this)
     );
+
+    this.emit('handshakeRequest', applicationName, authStrategy, credentials);
```

Comment:
@lundibundi, that's why I'm opting for not emitting them here: if a user will need to debug application including authentication he/she can easily do that by getting complete packets.
And outside of the debug mode emitting credentials makes no sense, no one should use them in that case, because logging them will lead to security problems (and you agree on this).